indexer/context/context.go

```diff
@@ -11,30 +12,62 @@ import (
 type IndexerContext interface {
 	Config() *config.Config
 	DB() *gorm.DB
+	Flags() *IndexerFlags
+}
+
+type IndexerFlags struct {
+	ConfigFileName string
+
+	// Set start epoch for voting cronjob to this value, overrides config and database value,
+	// valid value is > 0
+	ResetVotingCronjob int64
+
+	// Set start epoch for mirroring cronjob to this value, overrides config and database value,
+	// valid value is > 0
+	ResetMirrorCronjob int64
 }
 
 type indexerContext struct {
 	config *config.Config
 	db     *gorm.DB
+	flags  *IndexerFlags
 }
 
 func BuildContext() (IndexerContext, error) {
-	ctx := indexerContext{}
-
-	cfg, err := config.BuildConfig()
+	flags := parseIndexerFlags()
+	cfg, err := config.BuildConfig(flags.ConfigFileName)
 	if err != nil {
 		return nil, err
 	}
-	ctx.config = cfg
 	globalConfig.GlobalConfigCallback.Call(cfg)
 
-	ctx.db, err = database.ConnectAndInitialize(&cfg.DB)
+	db, err := database.ConnectAndInitialize(&cfg.DB)
 	if err != nil {
 		return nil, err
 	}
-	return &ctx, nil
+
+	return &indexerContext{
+		config: cfg,
+		db:     db,
+		flags:  flags,
+	}, nil
 }
 
 func (c *indexerContext) Config() *config.Config { return c.config }
 
 func (c *indexerContext) DB() *gorm.DB { return c.db }
+
+func (c *indexerContext) Flags() *IndexerFlags { return c.flags }
+
+func parseIndexerFlags() *IndexerFlags {
+	cfgFlag := flag.String("config", globalConfig.CONFIG_FILE, "Configuration file (toml format)")
```

Comment:
Just a general comment, personally I like using the `argparse` library for parsing command-line args, I prefer it to the builtin `flag` package. Not expecting you to change it here but just in case you want to check it out.